packages/material-ui-system/src/createStyled.d.ts

```diff
@@ -85,10 +85,13 @@ export interface StyledOptions {
   target?: string;
 }
 
-export interface MuiStyledOptions {
+export interface MuiStyledOptions<Props extends {}, ClassKey extends string | number | symbol> {
```

Comment:
Do we generally allow `number | symbol` as class keys? Historically, `ClassKey` was already used when using JSS and I think we always assumed strings back then and we still might refer to "ClassKey" internally as strings.